### ISSUE TITLE
SystemJS 4:  "Uncaught ReferenceError: err is not defined" in System.register in strict mode

### ISSUE THREAD

**Issue by stefanocke** [systemjs/systemjs#1965]
When upgrading from version 3.6.1 to 4.0.1, I get:
```
Uncaught ReferenceError: err is not defined
    at SystemJS.systemJSPrototype.register (system.js:476)
```

The regarding lines of code are:

```
  /*
   * Supports loading System.register via script tag injection
   */

  const systemRegister = systemJSPrototype.register;
  systemJSPrototype.register = function (deps, declare) {
    err = undefined;
    systemRegister.call(this, deps, declare);
  };
```

I has probably been caused by https://github.com/systemjs/systemjs/commit/1a5e951b8117d847a69093ada20cc08811957435
which removed a `let err` before that code block.

**Comment by joeldenning** [comment]
Hmm I believe that this error only occurs in strict mode, which is why we didn't catch it before releasing 4.0.1. The `err` variable was removed in #1961 but the reference inside of systemJSPrototype.register was forgotten about. And in strict mode you can't reassign variables that don't exist, which is something that you can do outside of strict mode.

Are you including systemjs into a bundle that is in strict mode? Or maybe running it in node where it has strict mode?

Either way, I've created #1966 with the fix.

**Comment by stefanocke** [comment]
@joeldenning , thank you.
Yes, it is in a bundle with some polyfills / shims and the bundle is in strict mode.

**Comment by joeldenning** [comment]
Fix is released in 4.0.2
